src/sync/list.rs

```diff
@@ -213,7 +239,7 @@ impl<'g, T: 'g, C: Container<T>> Iterator for Iter<'g, T, C> {
                 let succ = succ.with_tag(0);
 
                 match self.pred.compare_and_set_weak(
```

Comment:
How about using strong `compare_and_set` here?  We want to advance the global epoch as much as possible.